File: swing/src/main/java/wannabe/swing/renderer/FilledThreeDSquareWithCabinetSides.java
package wannabe.swing.renderer;

import java.awt.Graphics;
import wannabe.swing.SwingProjected;

public class FilledThreeDSquareWithCabinetSides extends SwingRenderer {
  Sides sides = new Sides();
  @Override public void draw(Graphics g, SwingProjected p) {
    sides.fillSides(g, p);
    if (!p.neighborAbove) g.fill3DRect(p.left, p.top, p.size, p.size, true);
  }
}

File: swing/src/main/java/wannabe/swing/renderer/FilledThreeDSquareWithCabinetSides.kt
package wannabe.swing.renderer

import wannabe.swing.SwingProjected
import java.awt.Graphics

class FilledThreeDSquareWithCabinetSides : SwingRenderer() {
  private var sides = Sides()

  override fun draw(
    g: Graphics,
    p: SwingProjected
  ) {
    sides.fillSides(g, p)
    if (!p.neighborAbove) g.fill3DRect(p.left, p.top, p.size, p.size, true)
  }
}
